### DISCUSSION TITLE
How to write meter plugin

### DISCUSSION THREAD

**Discussion by ouyangli** [apache/skywalking#8166] [Q&A] [answered]
https://skywalking.apache.org/docs/main/v8.7.0/en/guides/java-plugin-development-guide/ 
There is too little information about meter plugin.  How to customize a meter plugin? Is there a detailed development guide？

**Comment by wu-sheng** [comment, marked as answer]
This question is too general, but maybe this is your asking. https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/java-plugin-development-guide/#meter-plugin

You could take a reference from tracing plugin doc(same page) about how to intercept methods, and use meter APIs at the core level to collect numbers.

**Reply by ouyangli** [reply]
Counter counter = MeterFactory.counter("new_increase_count").tag("Order Count", "100").mode(CounterMode.INCREMENT).build();
            counter.increment(2d);

Why the summation and value fields of the meter_new_increase_count table are 0

**Reply by wu-sheng** [reply]
You need to build MAL script all received metrics. We would persistent unprocessed values to keep the cost of storage as less as possible

**Reply by ouyangli** [reply]
There is data in the database, and it is not 0, but the data in the dashboard is 0

![image](https://user-images.githubusercontent.com/8622409/142884369-760e1451-5150-4492-88cb-c047a7590036.png)

![image](https://user-images.githubusercontent.com/8622409/142884875-3f5d32b7-e4ad-4a97-a620-d83f1983fd52.png)


![image](https://user-images.githubusercontent.com/8622409/142885047-6003eb3a-2879-43aa-88bf-0b1fe639d886.png)

**Reply by wu-sheng** [reply]
You are looking at different time demonstrations, minutes VS days.

**Reply by ouyangli** [reply]
Thank you very much, the problem has been solved

**Comment by ouyangli** [comment]
![image](https://user-images.githubusercontent.com/8622409/143038985-d968fa12-4b50-4df4-abb3-2452d897b52c.png)
